src/wp-includes/fonts.php

```diff
@@ -128,44 +128,17 @@ function wp_get_font_dir() {
  * }
  */
 function wp_font_dir( $create_dir = true ) {
-	/*
-	 * Allow extenders to manipulate the font directory consistently.
-	 *
-	 * Ensures the upload_dir filter is fired both when calling this function
-	 * directly and when the upload directory is filtered in the Font Face
-	 * REST API endpoint.
-	 */
-	add_filter( 'upload_dir', '_wp_filter_font_directory' );
-	$font_dir = wp_upload_dir( null, $create_dir, false );
-	remove_filter( 'upload_dir', '_wp_filter_font_directory' );
-	return $font_dir;
-}
-
-/**
- * A callback function for use in the {@see 'upload_dir'} filter.
- *
- * This function is intended for internal use only and should not be used by plugins and themes.
- * Use wp_get_font_dir() instead.
- *
- * @since 6.5.0
- * @access private
- *
- * @param string $font_dir The font directory.
- * @return string The modified font directory.
- */
-function _wp_filter_font_directory( $font_dir ) {
-	if ( doing_filter( 'font_dir' ) ) {
-		// Avoid an infinite loop.
-		return $font_dir;
-	}
+	// This will also create the /uploads directory if it doesn't exist and $create_dir is true.
+	$upload_dir = wp_upload_dir( null, $create_dir, false );
 
+	// Default fonts directory settings.
 	$font_dir = array(
-		'path'    => untrailingslashit( $font_dir['basedir'] ) . '/fonts',
-		'url'     => untrailingslashit( $font_dir['baseurl'] ) . '/fonts',
+		'path'    => untrailingslashit( $upload_dir['basedir'] ) . '/fonts',
+		'url'     => untrailingslashit( $upload_dir['baseurl'] ) . '/fonts',
 		'subdir'  => '',
-		'basedir' => untrailingslashit( $font_dir['basedir'] ) . '/fonts',
-		'baseurl' => untrailingslashit( $font_dir['baseurl'] ) . '/fonts',
-		'error'   => false,
+		'basedir' => untrailingslashit( $upload_dir['basedir'] ) . '/fonts',
+		'baseurl' => untrailingslashit( $upload_dir['baseurl'] ) . '/fonts',
+		'error'   => $upload_dir['error'],
```

Comment:
Can remain false as it will be populated via the filter/wp_mkdir_p calls (related to not creating `uploads/yyy/mm` above)

```suggestion
		'error'   => false,
```

Reply:
> Can remain false as it will be populated via the filter/wp_mkdir_p calls

Yes, this error message may be populated by `wp_mkdir_p()` if the `$create_dir` param is true. however it will not be set when that param is false, right? 

What would (and what should) happen then there is an error returned by the `wp_upload_dir()` call above? Thinking that passing the original error and not trying to create a sub-directory when the parent directory is in an error state is the proper thing to do. This is actually one of the edge case fixes in this PR. 

Also, as mentioned, a plugin can set that error text and indicate the /uploads directory cannot be used by using the `upload_dir` filter. Thinking that `wp_font_dir()` should respect the returned value from `wp_upload_dir()` the same way as the rest of the code in core.

Reply:
You're right, the `_wp_upload_dir()` function sets this to false so it will exist even if the directory is not created.

`wp_upload_dir()` ignores the value and attempts directory creation regardless of the setting if `$create_dir` is true.

Reply:
> wp_upload_dir() ignores the value and attempts directory creation regardless of the setting if $create_dir is true.

Right, can't remember exactly but think that actually was a back-compat fix (or was it a bugfix) long time ago. We can repeat the same behavior here, and ignore errors coming from `wp_upload_dir()` and errors set by plugins in the `font_dir` filter. This is how `wp_upload_dir()` works.

However maybe better to respect these errors? Or at least respect an error that is set by a plugin and not attempt to create the directory?  What do you think?

Reply:
If the error is generated on by the `upload_dir` filter then I guess we can respect it. It would be an odd choice (for the plugin author) but sure.

The call to `wp_upload_dir()` shouldn't attempt to create a directory (which won't return an error without a plugin) as this code would be reacting to an error creating a directory that is unrelated to the one the code needs to create.